internal/btf/core.go

```diff
@@ -522,26 +564,74 @@ func adjustOffset(base uint32, t Type, n int) (uint32, error) {
 	return base + (uint32(n) * uint32(size) * 8), nil
 }
 
+func calculateBitfieldLoadJussi(_, bitOffset, bitSize int) (size int, offset int, err error) {
```

Comment:
Regarding the sizing algorithm, if there is a problem with `libbpf` one, maybe you should propose yours (can be done later nonetheless).

Reply:
There's multiple "working" ways to solving this. I'm not yet sure what's the "simplest" though (see comment below).